packages/block-editor/src/hooks/layout.js

```diff
@@ -313,23 +313,28 @@ export default {
 	},
 };
 
-function LayoutTypeSwitcher( { type, onChange } ) {
+function LayoutTypeSwitcher( { key, onChange } ) {
 	return (
-		<ButtonGroup>
+		<ToggleGroupControl
+			__next40pxDefaultSize
+			isBlock
+			label={ __( 'Layout Type Select' ) }
+			__nextHasNoMarginBottom
+			hideLabelFromVision
+			isAdaptiveWidth
+		>
 			{ getLayoutTypes().map( ( { name, label } ) => {
 				return (
-					<Button
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
-						key={ name }
-						isPressed={ type === name }
+					<ToggleGroupControlOption
+						value={ name }
+						label={ name }
```

Comment:
Looks like the component APIs are used a bit incorrectly. We'll need the following changes to keep it in sync with the block variation, and for the options to have the correct labels.

```diff
diff --git a/packages/block-editor/src/hooks/layout.js b/packages/block-editor/src/hooks/layout.js
index f437a82ac1..90859e5e43 100644
--- a/packages/block-editor/src/hooks/layout.js
+++ b/packages/block-editor/src/hooks/layout.js
@@ -313,7 +313,7 @@ export default {
 	},
 };
 
-function LayoutTypeSwitcher( { key, onChange } ) {
+function LayoutTypeSwitcher( { type, onChange } ) {
 	return (
 		<ToggleGroupControl
 			__next40pxDefaultSize
@@ -322,16 +322,16 @@ function LayoutTypeSwitcher( { key, onChange } ) {
 			__nextHasNoMarginBottom
 			hideLabelFromVision
 			isAdaptiveWidth
+			value={ type }
+			onChange={ onChange }
 		>
 			{ getLayoutTypes().map( ( { name, label } ) => {
 				return (
 					<ToggleGroupControlOption
+						key={ name }
 						value={ name }
-						label={ name }
-						onClick={ () => onChange( name ) }
-					>
-						{ label }
-					</ToggleGroupControlOption>
+						label={ label }
+					/>
 				);
 			} ) }
 		</ToggleGroupControl>

```

https://github.com/user-attachments/assets/5c9c758b-f448-4086-8833-526c6cd11853

Reply:
@mirka Thank you for the feedback. I'm still green when it comes to the components. I made the changes you suggested. Please take a look when you get a chance. https://github.com/WordPress/gutenberg/pull/65498/commits/55ec4690bd2b13871f6118365fdb6b30b49b6316

Reply:
No worries, they can be complicated 😆